File: Cine/gestorAplicación/tiendaAbst/TiendaUN.java
package gestorAplicación.tiendaAbst;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;
import gestorAplicación.cine.Usuario;

public class TiendaUN extends Tienda implements Serializable{
	//atributos
	private static final long serialVersionUID = 1L;

    public TiendaUN(String nombre) {
        super(nombre);
        
    }

    protected Map<String, Integer> inventario = new HashMap<String, Integer >();
    protected Map<String, Integer> productos = new HashMap<String, Integer >();


    //constructor
    public void venderProducto(String nombre, Integer cantidad, Usuario usuario) {
        if (productos.get(nombre) >= cantidad) {
            productos.put(nombre, productos.get(nombre) - cantidad);
            if (usuario.verificarMembresia()) {
                usuario.setSaldo(usuario.getSaldo() - (int)((cantidad * inventario.get(nombre))*0.3) );
            }
            else {
                usuario.setSaldo(usuario.getSaldo() - (cantidad * inventario.get(nombre)));
            }
        }
    }
//metodos get y set
    public Map<String, Integer> getInventario() {
        return inventario;
        
    }

    public void setInventario(Map<String, Integer> inventario) {
        this.inventario = inventario;
        
    }

    public Map<String, Integer> getProductos() {
        return productos;
        
    }

    public void setProductos(Map<String, Integer> productos) {
        this.productos = productos;
        
    }

    public void agregarProducto(String nombre, Integer cantidad) {
        productos.put(nombre, cantidad);
        
    }

    public void agregarInventario(String nombre, Integer cantidad) {
        inventario.put(nombre, cantidad);
        
    }

    public void agregarMixto (String nombre, Integer cantidadProducto, Integer precioDelProducto) {
        productos.put(nombre, cantidadProducto);
        inventario.put(nombre, precioDelProducto);
        
    }

    public int getPrecio(String nombre) {
        return productos.get(nombre);
        
    }
// sobreescritura de metodo abstract
    @Override
    public void saludo() {
        System.out.println("Bienvenido a la tienda del cine UNAL");
        
    }
    
}

File: Python/src/gestionAplicacion/tiendapy/tiendaUN.py
from gestionAplicacion.tiendapy.Tienda import Tienda


class TiendaUN(Tienda):

    tiendaUN = []
    # inventario is a dictionary of String:Integer
    # productos is a dictionary of String:Integer
    def __init__(self, nombre, inventario, productos):
        super.__init__(self, nombre, inventario, productos)
       
    def getNombre(self):
        return self.nombre

    def getInventario(self):
        return self.inventario

    def getProductos(self):
        return self.productos

    def setNombre(self, nombre):
        self.nombre = nombre

    def setInventario(self, inventario):
        self.inventario = inventario

    def setProductos(self, productos):
        self.productos = productos

    def venderProducto(self, producto):
        if (self.productos[producto] > 0):
            if (usuario.getSaldo() >= self.productos[producto]):
                if (usuario.isVip()):
                    self.productos[producto] = self.productos[producto] - 1
                    usuario.setSaldo(usuario.getSaldo() - self.productos[producto]*0.8)
                    print("Gracias por su compra")
            else:
                print("No tiene suficiente dinero")
        else:
            print("No hay suficientes productos")

def agregarMixto(self, producto, cantidadUnidades, precio):
        self.productos[producto] = precio
        self.inventario[producto] = cantidadUnidades
    
def __str__(self):
    return "Bienvenido a la tienda de la UN"

def getTiendaUN():
    return Tienda.tienda

def setTiendaUN(tiendaUN):
    Tienda.tienda.append(tiendaUN)

